diff --git a/src/EasySVG.php b/src/EasySVG.php
index a21e5b0..1bda23c 100644
--- a/src/EasySVG.php
+++ b/src/EasySVG.php
@@ -390,6 +390,10 @@ public function unicodeDef(string $unicode): string
 
         // extract character definition
         $d = $this->font->glyphs[hexdec($unicode)]->d;
+        // handle if d attribute is not set in corresponding glyph
+        if(is_null($d)) {
+            $d = "";
+        }
 
         // transform typo from original SVG format to straight display
         $d = $this->defScale($d, $fontSize, -$fontSize);
